assets/src/edit-story/components/dropTargets/karma/background.karma.js

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { Fixture } from '../../../karma';
+import { useStory } from '../../../app/story';
+import { useInsertElement } from '../../../components/canvas';
+
+describe('Background Drop-Target integration', () => {
+  let fixture;
+
+  beforeEach(async () => {
+    fixture = new Fixture();
+    await fixture.render();
+    jasmine.addMatchers(customMatchers);
+  });
+
+  afterEach(() => {
+    fixture.restore();
+  });
+
+  it('should by default have white background', async () => {
+    const bgElement = await getCanvasBackgroundElement(fixture);
+    // Verify that it's empty
+    expect(bgElement).toBeEmpty();
+    // And that background color is white:
+    expect(bgElement).toHaveBackgroundColor('rgb(255, 255, 255)');
+  });
+
+  describe('when there is an image on the canvas', () => {
+    let imageData;
+
+    beforeEach(async () => {
+      imageData = await addDummyImage(fixture);
+      jasmine.DEFAULT_TIMEOUT_INTERVAL = 1000000000;
+    });
+
+    it('should correctly handle image dropped on edge', async () => {
+      const backgroundId = await getBackgroundElementId(fixture);
+
+      // Verify that bg replacement is empty
+      const rep1 = await getCanvasBackgroundReplacement(fixture);
+      expect(rep1).toBeEmpty();
+
+      // Drag the image element to the background
+      await dragToDropTarget(fixture, imageData.id, backgroundId);
+
+      // Verify that bg replacement is no longer empty
+      expect(rep1).not.toBeEmpty();
+
+      // Verify that replacement img has correct source
+      const replaceImg = rep1.querySelector('img');
+      expect(replaceImg).toHaveSource(imageData.resource.src);
+
+      // Now drop the element
+      await fixture.events.mouse.up();
+
+      // Verify new background element has the correct image
+      const bg = await getCanvasBackgroundElement(fixture);
+      const bgImg = bg.querySelector('img');
+      expect(bgImg).toHaveSource(imageData.resource.src);
+
+      // And verify that we no longer have a replacement element
+      const rep2 = await getCanvasBackgroundReplacement(fixture);
+      expect(rep2).toBeEmpty();
+    });
```

Comment:
I feel this reads very well now, but it required quite a few helper functions and custom matchers to be this compact. This definitely has potential though - and runs quickly!